controllers/profiles/reconciler_dev.go

```diff
@@ -129,12 +160,20 @@ func (e *ensureRunningDevWorkflowReconciliationState) Do(ctx context.Context, wo
 	}
 	objs = append(objs, deployment)
 
-	service, _, err := e.ensurers.service.ensure(ctx, workflow, defaultServiceMutateVisitor(workflow))
+	service, _, err := e.ensurers.service.ensure(ctx, workflow, defaultServiceMutateVisitor(workflow), devProfileServiceMutateVisitor(workflow))
 	if err != nil {
 		return ctrl.Result{RequeueAfter: requeueAfterFailure}, objs, err
 	}
 	objs = append(objs, service)
 
+	if e.ensurers.network != nil {
```

Comment:
Instead, apply the "null pattern" here. In the creator, we can have an ensurer that does nothing and just return. We avoid adding conditionals to the code.

Reply:
I applied the null pattern as suggested ;-)